### ISSUE TITLE
Problem with magma_dstedx()

### ISSUE THREAD

**Issue by AhmadObeid** [hcho3/eigenmap_gpu#2]
When I make the MakeFile and it reaches the objective line lanczos.o it reports a problem. 
Namely, in line 123 of lanczos.cu, the function magma_dstedx() has the first input set as a character as shown here: 
`magma_dstedx('I', LANCZOS_ITR, 0, 1, 1, LANCZOS_ITR, &alpha[1],
                 &beta[1], eigvec, LANCZOS_ITR, work, lwork, iwork,
                 liwork, dwork, &info);`
and this gives the following error:

> lanczos.cu(123): error: argument of type "char" is incompatible with parameter of type "magma_range_t"
> 
> 1 error detected in the compilation of "/tmp/tmpxft_00005bbf_00000000-6_lanczos.cpp1.ii".
> Makefile:50: recipe for target 'lanczos.o' failed.

Any idea?

Thank you,

Ahmad

**Comment by chyunsu3** [comment]
@AhmadObeid It looks like the signature of the `magma_dstedx()` function was changed when MAGMA version was upgraded from 1.4.1 to 1.5.0. I wrote this code from 2013 to 2014, so it is not compatible with later versions of MAGMA. Please install MAGMA 1.4.1.

**Comment by AhmadObeid** [comment]
I am getting close to having it work, so thank you for your answer. However another error popped up after I installed MAGMA 1.4.1. This time it tells me: 
 

> #error "One of HAVE_CUBLAS, HAVE_clAmdBlas, or HAVE_MIC must be defined. This typically happens in Makefile.internal."

I tried implementing the solution suggested [here](http://icl.cs.utk.edu/magma/forum/viewtopic.php?f=2&t=1065&p=2918&hilit=underscore) by hsahasra, by adding the lines he mentioned in the file lanczos.cu, right before I include magma.h. Unfortunately the error persists. 

Thank you again for your help.

Ahmad
